### ISSUE TITLE
Admin hangs at migration after upgrade from 2.0.28 to 2.0.29

### ISSUE THREAD

**Issue by DerKnerd** [Mailu/Mailu#2992]
### Environment

- [x] docker compose
- [ ] kubernetes
- [ ] docker swarm

### Version

- Version: `2.0.29`

## Description

After upgrading my setup from version 2.0.28 to 2.0.29 the admin container hangs during startup at the following messages:

```
INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
```

Changing the log level to DEBUG doesn't have any effect.

## Replication Steps

Take a base installation of Mailu with data in the database and upgrade it from 2.0.28 to 2.0.29.

Greetings
Imanuel

There is another issue in Mailu/helm-charts for this, but I am not using the helm chart. I wrote the manifests based on the docker-compose.yaml by myself. So I created another issue here.

Here is the helm charts issue: https://github.com/Mailu/helm-charts/issues/309

**Comment by DerKnerd** [comment]
Same with version 2.0.30.

**Comment by DerKnerd** [comment]
@nextgens could you elaborate?

**Comment by nextgens** [comment]
Yes.

**Comment by nextgens** [comment]
I don't see how it could be made more explicit:
https://github.com/Mailu/Mailu/blob/efcf7a1581c92e6a0e57462b12f8d86f4b4b0a1c/ISSUE_TEMPLATE.md?plain=1#L33

You are knowingly running Mailu in an unsupported way; I have helpfully moved your ticket for you the first time around and now you are opening another... You are free to run the software in whatever way you want, just don't expect support for doing it.

We are dedicating the scant resources of the project towards supporting what we actually say is supported, that is running our set of docker images using dockerd.

You can thank @Esterinox for the change in policy.

**Comment by Elrondo46** [comment]
What kind of TOXIC project stigmatize a user/contributor exposed an issue... this project MUST BE FORKED to become detoxified

**Comment by Elrondo46** [comment]
@Esterinox solved his issue by himself and I think we will fork this project to have a human respectfull project

**Comment by DerKnerd** [comment]
After some research, it seems to be related to the generation of migrations during startup.

**Comment by pr0ton11** [comment]
@DerKnerd Look at this commit https://github.com/Mailu/Mailu/commit/2a570d0f6f9f5d799ffcaaf68963e729830e03b9#diff-55df9d2f8b8d40391d9d451a1e8155292465d48c52e78dd34655be834cd3dc9fR886

@nextgens Could you make sure that the log message does actually show up? I wasted hours troubleshooting exactly this issue because the log message is not shown. Also I don't think breaking working setups to prevent people from creating issues with unsupported setups is worth it, as in my case it did exactly the opposite. I had to open an issue to figure out what is preventing the admin container from starting.

**Comment by Esterinox** [comment]
I'd like this project to be more open-minded. 
Docker is disappearing, you'll have more people using Podman as time goes by. I think it's a shame that every time someone uses Podman, they get their ISSUE closed without a response. The only excuse is "It's not officially supported by mailu", but that's just being hyper closed-minded.
On the contrary, native support for these new technologies would only benefit the project. There's absolutely nothing negative about it. 
When I make an Issue I'm always very cordial and I get it swept away like that for free, so don't be surprised if someone reacts badly. 
There's nothing human left in free software, it's a shame to see such degradation. It's a shame to see such degradation. We could all talk to each other in a calm manner, so that all requests are taken seriously and we can move forward together. 
That's all I'm asking. A project at the heart of the debate, so that we can move forward efficiently without needlessly taking each other's heads off.

**Comment by nextgens** [comment]
> @nextgens Could you make sure that the log message does actually show up?

Sure but show up where? The released build uses stdout, master the logging system as the other commit hasn't been backported.

> I wasted hours troubleshooting exactly this issue because the log message is not shown. Also I don't think breaking working setups to prevent people from creating issues with unsupported setups is worth it, as in my case it did exactly the opposite. I had to open an issue to figure out what is preventing the admin container from starting.

I am ready to hear that what has been done is counter productive but, do you have a constructive way of solving the problem to suggest? We have it [documented in the FAQ](https://mailu.io/2.0/faq.html#how-can-i-use-mailu-without-docker)! 

I think that the next course of action is to actually restrict the submission of issues to existing contributors and push everything else to discussions.

**Comment by pr0ton11** [comment]
So I guess this would be fixed in the next release so the message actually shows up in the log (e.g when running docker / podman logs)? Then thats fine for me and I understand that this is a unsupported configuration that the resources of the project are limited.

About the question for issues I can not really contribute any easy solution. I would suggest backporting the fix as fast as possible to limit this problem from showing up again.

Also maybe keep in mind that for bugfix and minor releases I would not expect breaking changes like these are added. And while this might have been a reaction to people constantly asking about unsupported platforms with requests unrelated to mailu it also affects users like me who generally try to not bother devs with unnecessary/unsupported requests.

**Comment by Esterinox** [comment]
"This comment was marked as abuse."
What's wrong with my comment?
I'm really not appreciated here...

**Comment by ghostwheel42** [comment]
@Esterinox As I wrote in another thread: I'm using podman and kubernetes for other things, but I'm using docker-compose to run Mailu, as this is the way it was designed to run.
When people join willing to help supporting / developing other platforms/features Mailu probably can move away from docker (I thought about Quadlet in another discussion). Everything is possible, but there need to be more people willing to contribute - especially when more than one platform is to be supported.

**Comment by nextgens** [comment]
@pr0ton11 Mailu is likely to become more and more tightly integrated as it evolves. The value of using something like Mailu is to run a stack that has been tested and fine tuned by others. If you do not need the integration you can "roll out everything on your own".

Unfortunately we can't do "advanced" things in a platform agnostic way without a lot more resources and the few times where we have attempted it we were bitten back by reality. I can give you a handful of examples on top of my head where what looked like a simple change has proved to be problematic outside of the dockerd realm. The first one that springs to mind would be DNSSEC related things when we decided to introduce DANE validation: expecting that people will operate an email server with a functioning stack that can do DNS resolution properly... That turned out to be a massive undertaking with broken pieces all over the echo-system. https://mailu.io/master/faq.html#the-admin-container-won-t-start-and-its-log-says-critical-your-dns-resolver-isn-t-doing-dnssec-validation ... coredns (k8s), old versions of podman (with a dnsmasq without ``proxy-dnssec``), netplan ... and even recently we had to [upgrade our libc](https://github.com/Mailu/Mailu/issues/2934#issuecomment-1751648821) because of this.

I know that in the future we will be exploring how to make ipv6 work properly... that too seems very complicated, especially if we aim to retain podman compatibility (see #2293).

Nothing works in the real world and you will find an endless supply of users arguing that their way of doing things "is good enough/better" as "they have never noticed any problem". I will let you make your own mind up on what the parallel thread is about here: https://github.com/Mailu/Mailu/discussions/3017#discussioncomment-7414702 ... surely running an open-relay has nothing to do with the symptoms being exhibited... and no relation whatsoever with the undisclosed "customizations" that have been done. All good since it has been done in the name of Security. Right?

The fact is in the last 24h we had [4 new issues](https://github.com/Mailu/Mailu/issues?q=is%3Aissue) raised, none of which actually affect people running the current version of the software in the documented way. Something does need to be done.

**Comment by pr0ton11** [comment]
I generally agree, especially when security is a concern. I think the problem is also that that the target demographic of mailu is people who want to run a simple mail server and don't understand the concepts and/or security considerations.

Thats why documentation is a primary concern. But with the willingness of people to read documentation (and especially follow it properly) this will not be a solution to this specific problem.

I understand the dilemma, but in my personal case, I know what I am doing and simply would prefer not breaking my existing deployments. There must be another middleground way, e.g maybe instead provide setup scripts that install dependencies show a warning that the user has to confirm and then deploy the stack, e.g with a few supported distros. Then only support installations done with this script.

I used podman in the first place because I come from a RHEL background where docker is no longer an option.

**Comment by paulcsiki** [comment]
I was impacted by this as well and while I agree with the approach I would have mentioned in the error message what exactly is wrong and perhaps a link to the FAQ on how to unblock it. Looking at the source code I was presented with obfuscated strings for some reason which I don't understand. I agree that something needs to be done with custom deployments but at least make it obvious on what needs to be done. Read the documentation is not read the FAQ.
